pkg/executor/analyze_utils.go

```diff
@@ -24,11 +24,46 @@ import (
 	"github.com/pingcap/tidb/pkg/sessionctx"
 	"github.com/pingcap/tidb/pkg/sessionctx/variable"
 	"github.com/pingcap/tidb/pkg/statistics"
+	"github.com/pingcap/tidb/pkg/store/helper"
 	"github.com/pingcap/tidb/pkg/util"
+	"github.com/pingcap/tidb/pkg/util/logutil"
 	"github.com/tiancaiamao/gp"
 	"go.uber.org/atomic"
+	"go.uber.org/zap"
 )
 
+func adaptiveAnlayzeDistSQLConcurrency(ctx context.Context, sctx sessionctx.Context) int {
```

Comment:
How often is this function called? Could it be invoked so frequently in some cases that it overwhelms the PD API server?

Reply:
Every time, analyze jobs need to run.

Reply:
> How often is this function called? Could it be invoked so frequently in some cases that it overwhelms the PD API server?

Auto analyze. But I think it's QPS <= 10 qps